pkg/controller/validate/validate_test.go

```diff
@@ -24,35 +24,35 @@ import (
 func TestClusterValidation(t *testing.T) {
 	t.Run("Invalid cluster specs", func(t *testing.T) {
 		t.Run("Multiple specs specified", func(t *testing.T) {
-			spec := mdbv1.AtlasDeploymentSpec{AdvancedDeploymentSpec: &mdbv1.AdvancedDeploymentSpec{}, DeploymentSpec: &mdbv1.DeploymentSpec{}}
+			spec := mdbv1.AtlasDeploymentSpec{DeploymentSpec: &mdbv1.AdvancedDeploymentSpec{}, ServerlessSpec: &mdbv1.ServerlessSpec{}}
 			assert.Error(t, DeploymentSpec(&spec, false, "NONE"))
 		})
 		t.Run("No specs specified", func(t *testing.T) {
-			spec := mdbv1.AtlasDeploymentSpec{AdvancedDeploymentSpec: nil, DeploymentSpec: nil}
-			assert.Error(t, DeploymentSpec(&spec, false, "NONE"))
-		})
-		t.Run("Instance size not empty when serverless", func(t *testing.T) {
-			spec := mdbv1.AtlasDeploymentSpec{AdvancedDeploymentSpec: nil, DeploymentSpec: &mdbv1.DeploymentSpec{
-				ProviderSettings: &mdbv1.ProviderSettingsSpec{
-					InstanceSizeName: "M10",
-					ProviderName:     "SERVERLESS",
-				},
-			}}
-			assert.Error(t, DeploymentSpec(&spec, false, "NONE"))
-		})
-		t.Run("Instance size unset when not serverless", func(t *testing.T) {
-			spec := mdbv1.AtlasDeploymentSpec{AdvancedDeploymentSpec: nil, DeploymentSpec: &mdbv1.DeploymentSpec{
-				ProviderSettings: &mdbv1.ProviderSettingsSpec{
-					InstanceSizeName: "",
-					ProviderName:     "AWS",
-				},
-			}}
+			spec := mdbv1.AtlasDeploymentSpec{DeploymentSpec: nil}
 			assert.Error(t, DeploymentSpec(&spec, false, "NONE"))
 		})
+		// t.Run("Instance size not empty when serverless", func(t *testing.T) {
```

Comment:
These tests looks odd, while it says to be about serverless the code doesn't reflect that. I think it can be removed